### PR TITLE
 Created a way to interact with our library without requiring the user to `import "gl-matrix";`

### LINKED ISSUE #48: Better user-facing API for vectors
We probably don't want users to have to write `someArmature.point(vec3.fromValues(0, 0, 0))`. I think allowing a syntax like `.point({x: 0, y: 0, z: 0})` might be better, and then we can convert behind the scenes.

### PR DESCRIPTION
**Pull request by armcburney**
Fixes #48.

### Todo
- [x] Merge in #86 before this PR
- [x] Test changes locally with `yarn webpack` once the build is fixed
- [x] More documentation

### CODE REVIEW

**Review by davepagurek — APPROVED** [review]
Nice! this is great, goes a long way towards making this library actually useable

**Comment by armcburney** [comment]
Hey @davepagurek, do you know why the build is failing due to:

```
/home/circleci/repo/src/armature/Node.ts:386:30
ERROR: 386:30  no-use-before-declare  variable 'Point' used before declaration
```

I have the disable in `Node.ts` line 385, so I'm confused. 😳

**Comment by davepagurek** [comment]
Yeah, circular class dependencies are unavoidable sometimes. I used `// tslint:disable-next-line:no-use-before-declare` somewhere before

File: src/examples/generation.ts [hunk 1]
```diff
@@ -55,13 +58,11 @@ const tree = treeGen
...
-        node.setRotation(
-            mat4.fromQuat(mat4.create(), quat.fromEuler(quat.create(), theta, phi, 0))
-        );
-        node.setScale(mat4.fromScaling(mat4.create(), vec3.fromValues(0.8, 0.8, 0.8))); // Shrink a bit
+        node.setRotation(Matrix.fromQuat4(Quaternion.fromEuler(theta, phi, 0)));
+        node.setScale(Matrix.fromScaling4({ x: 0.8, y: 0.8, z: 0.8 })); // Shrink a bit
 
         const trunk = node.point('mid').attach(branchSphere);
-        trunk.setScale(mat4.fromScaling(mat4.create(), vec3.fromValues(0.2, 0.8, 0.2)));
+        trunk.setScale(Matrix.fromScaling4({ x: 0.2, y: 0.8, z: 0.2 }));
```
Comment (davepagurek) [inline src/examples/generation.ts]: Can we call this just `.fromScaling`?

**Review by davepagurek — APPROVED** [review]

**Pull request merged**